core/src/main/java/jenkins/model/ProjectNamingStrategy.java

```diff
@@ -99,6 +101,11 @@ public DefaultProjectNamingStrategy() {
         @Override
         public void checkName(String origName) throws Failure {
             // default - should just do nothing (this is how Jenkins worked before introducing this ExtensionPoint)
+            if (StringUtils.isNotBlank(DEFAULT_PATTERN) && StringUtils.isNotBlank(origName)) {
```

Comment:
imho, check for blank string -> throw blank error, check for single quote -> throw single quote. As job tied to FS probably this should be done with some additional restrictions like `/`